pkg/genutil/slicez/slicez.go

```diff
@@ -0,0 +1,41 @@
+package slicez
```

Comment:
This is awesome. I'm kinda surprised it's not somewhere in the stdlib yet 🤔

Reply:
lots of stuff missing for iterators too: https://github.com/jzelinskie/itz/blob/main/itz.go